js/runtime.js

```diff
@@ -1125,6 +1125,24 @@
                 var date = new Date(prevDate.valueOf());
                 date.setFullYear(date.getFullYear() + years);
                 return date;
+            },
+            dayOfWeek: function(date) {
+                var days = ['Sunday', 'Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday'];
+                return days[date.getDay()];
+            },
+            getDay: function(date) {
+                return date.getDate();
+            },
+            getMonth: function(date) {
```

Comment:
Might want to have getMonth() return a number for using in other code and getMonthName() return a string? I was hoping we could get the text name of the month from the Date object itself, so we would get the right language for the users's locale, but apparently that is actually pretty tough to do cross-platform. More detals https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/toLocaleDateString and https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/DateTimeFormat (and I couldn't find in either of those how we would even *get* the user's locale).